### PR TITLE
Univariate root finding

### PR DESCRIPTION
**Pull request by alex-ozdemir**

### CODE REVIEW

File: src/theory/ff/roots.cpp [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+template <typename T>
+std::string sstring(const T& t)
+{
+  std::ostringstream o;
+  o << t;
+  return o.str();
+}
+
+// sorting based on strings because CoCoA doesn't cmp
```
[the remark this answers is not in the document]
Comment (alex-ozdemir) [reply src/theory/ff/roots.cpp]: Yes, that's a natural question. Unfortunately, comparison only works for elements of 'ordered' rings: i.e., a ring `R` for which `CoCoA::IsOrderedDomain(R)` returns true. Most rings that CoCoA implements do not satisfy this, including prime fields.

I've expanded the comment a bit.

**Comment by alex-ozdemir** [comment]
Following up on our discussion, the global manager must be created *exactly* once.

Creating it multiple times does cause an error.

**Review by ajreynol — CHANGES_REQUESTED** [review]
Just a request to doc the singleton pointer.

File: src/util/cocoa_globals.h [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+#ifdef CVC5_USE_COCOA
+
+#ifndef CVC5__UTIL__COCOA_GLOBALS_H
+#define CVC5__UTIL__COCOA_GLOBALS_H
+
+#include <CoCoA/GlobalManager.H>
+
+namespace cvc5::internal {
+
+extern CoCoA::GlobalManager* s_cocoaGlobalManager;
```
[the remark this answers is not in the document]
Comment (alex-ozdemir) [reply src/util/cocoa_globals.h:40]: Yep! Just did it. I put most of the documentation at the top of the header.

**Review by ajreynol — APPROVED** [review]
LGTM

**Pull request merged**